contracts/BMTFile.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./BMTChunk.sol";
+
+
+contract BMTFile is BMTChunk {
+
+// max segment count
+  uint256 public constant MAX_SEGMENT_COUNT = 128;
+
+  // chunk bmt levels
+  uint256 public constant CHUNK_BMT_LEVELS = 7;
+
+  struct ChunkInclusionProof{
+    uint64 span;
+    bytes32[] sisterSegments;
+  }
+
+  /* 
+  * Gives back the file address that is calculated with only the inclusion proof segments
+  * and the corresponding proved segment and its position.
+  * @param _proveChunks Sister segments that will be hashed together with the calculated hashes
+  * @param _proveSegment The segment that is wanted to be validated it is subsumed under the file address
+  * @param _proveSegmentIndex the `proveSegment`'s segment index on its BMT level
+  * @param _fileLength File length
+  * @return _calculatedHash File address
+  */
+function fileAddressFromInclusionProof(
+  ChunkInclusionProof[] memory _proveChunks,
+  bytes32 _proveSegment,
+  uint256 _proveSegmentIndex,
+  uint256 _fileLength
+) public pure returns (bytes32 _calculatedHash) {
+   _calculatedHash = _proveSegment;
+
+  for (uint256 i = 0; i < _proveChunks.length; i++) {
+    (uint256 parentChunkIndex, uint256 level) = getBmtIndexOfSegment(
+      _proveSegmentIndex,
+      _fileLength
```

Comment:
here you ask for the `_lastChunkIndex` and you pass the `_fileLength` instead..

consider these snippet so that you can rid of this additional `_fileLength` argument, as it is already given in the `_proveChunks` array:

https://github.com/fairDataSociety/bmt-js/blob/1ca345af958d49d95359356c1881950b8f9d7838/src/file.ts#L154-L155

Reply:
not necessary to do that in the same way, you can shift the bits as well instead